app/scripts/services/validateRegistrant.js

```diff
@@ -4,52 +4,36 @@ angular.module('confRegistrationWebApp')
   .service('validateRegistrant', function validateRegistrant() {
 
     var blockVisibleRuleCheck = function (block, registrant) {
-      var returnValueAND = true;
-      var returnValueOR = false;
       var answers = registrant.answers;
-      var ruleOperand = block.content && block.content !== '' && !angular.isUndefined(block.content.ruleoperand) ? block.content.ruleoperand : 'AND';
+      var ruleOperand = block.content && block.content.ruleoperand ? block.content.ruleoperand : 'AND';
+      var ruleArray = [];
 
-      if (ruleOperand === 'AND') {
-        angular.forEach(block.rules, function (rule) {
-          var answer = _.find(answers, { blockId: rule.parentBlockId });
```

Comment:
@OzzieOrca It's not about not liking your code. We just thought that this new piece of code snippet could do the job in fewer lines and is more legible - at least better than the previous code we came up with. If you don't approve, we are ready to change it to suit your need.